cpp/src/gandiva/jni/jni_common.cc

```diff
@@ -750,8 +750,8 @@ JNIEXPORT jlong JNICALL Java_org_apache_arrow_gandiva_evaluator_JniWrapper_build
   return module_id;
 }
 
-JNIEXPORT jint JNICALL Java_org_apache_arrow_gandiva_evaluator_JniWrapper_evaluateFilter(
-    JNIEnv* env, jobject cls, jlong module_id, jint num_rows, jlongArray buf_addrs,
+JNIEXPORT jlong JNICALL Java_org_apache_arrow_gandiva_evaluator_JniWrapper_evaluateFilter(
```

Comment:
changing this would mean you would need to change the jni declaration too, [here](https://github.com/apache/arrow/blob/master/java/gandiva/src/main/java/org/apache/arrow/gandiva/evaluator/JniWrapper.java#L190)
which is why you are getting an unsatisfied link error while running the java tests..